### PR TITLE
HIP-584: Change InvalidTransactionException type in Store

### PR DESCRIPTION
**Pull request by bilyana-gospodinova**
**Description**:
When the store does not find an object and the OnMissing parameter is set to THROW an InvalidTransactionException is thrown (from the web3 package). In HTSPrecompiledContract#computeInternal we catch InvalidTransactionException from service so in this case the exception is caught in the general Exception block, the frame does not revert and the response is successful although it is not. Junit tests are adapted.

Fixes https://github.com/hashgraph/hedera-mirror-node/issues/6787

### CODE REVIEW

**Review by 0xivanov — COMMENTED** [review]
You can add new negative test in ContractCallDynamicCallsTest.java where we pass not persisted address as parameter (the transaction should fail in the store) and assert that the transaction reverts.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/store/StoreImplTest.java [hunk 1]
```diff
@@ -39,7 +39,6 @@
 import com.hedera.mirror.web3.evm.store.accessor.TokenRelationshipDatabaseAccessor;
 import com.hedera.mirror.web3.evm.store.accessor.UniqueTokenDatabaseAccessor;
 import com.hedera.mirror.web3.evm.store.accessor.model.TokenRelationshipKey;
-import com.hedera.mirror.web3.exception.InvalidTransactionException;
```
[the remark this answers is not in the document]
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/store/StoreImplTest.java:42]: The exception was renamed to MirrorEvmTransactionException.

[the remark this answers is not in the document]
Comment (0xivanov) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/store/StoreImplTest.java:42]: I vote for `MirrorEvmInvalidTransactionException` :D

File: hedera-mirror-web3/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java [hunk 2]
```diff
@@ -1063,9 +1063,9 @@ void noAssociationsWithMissingAccounts() {
 
     @Test
     void applicationRejectsMissing() {
-        given(store.getToken(asTypedEvmAddress(misc), OnMissing.THROW)).willThrow(InvalidTransactionException.class);
+        given(store.getToken(asTypedEvmAddress(misc), OnMissing.THROW)).willThrow(MirrorEvmTransactionException.class);
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java:1066]: Believe this should be the services InvalidTransactionException

File: hedera-mirror-web3/src/test/java/com/hedera/services/txn/token/AssociateLogicTest.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
 import com.hedera.mirror.web3.evm.store.Store.OnMissing;
 import com.hedera.mirror.web3.evm.store.StoreImpl;
 import com.hedera.mirror.web3.evm.store.accessor.model.TokenRelationshipKey;
-import com.hedera.mirror.web3.exception.InvalidTransactionException;
+import com.hedera.mirror.web3.exception.MirrorEvmTransactionException;
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/services/txn/token/AssociateLogicTest.java:31]: This should be the services InvalidTransactionException

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Pull request merged**